### PR TITLE
Small post Rails 6.1 fixes

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

I ran `bundle install` after #13280 and ran `bundle exec rails db:migrate` after we upgraded to Rails 6.1.

See inline comments for further info

## Related Tickets & Documents

#13280
#11333 

## Added tests?

- [ ] Yes
- [x] No, and this is why: not needed
- [ ] I need help with writing tests

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -842,7 +842,7 @@ GEM
 
 PLATFORMS
   arm64-darwin-20
-  x86_64-darwin-18 
+  x86_64-darwin
```
Comment (rhymes) [inline Gemfile.lock:845]: This is what `bundle install` on macOS Catalina left me with

File: db/schema.rb [hunk 2]
```diff
@@ -2,8 +2,8 @@
 # of editing this file, please use the migrations feature of Active Record to
 # incrementally modify your database, and then regenerate this schema definition.
 #
-# This file is the source Rails uses to define your schema when running `rails
-# db:schema:load`. When creating a new database, `rails db:schema:load` tends to
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
```
Comment (rhymes) [inline db/schema.rb:6]: this is changed automatically by Rails 6.1

File: db/schema.rb [hunk 3]
```diff
@@ -1350,6 +1350,7 @@
     t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
     t.index ["twitter_username"], name: "index_users_on_twitter_username", unique: true
     t.index ["username"], name: "index_users_on_username", unique: true
+    t.check_constraint "username IS NOT NULL", name: "users_username_not_null"
```
Comment (rhymes) [inline db/schema.rb:1353]: Rails 6.1 finally dumps check constraints, see https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/Table.html#method-i-check_constraint

@citizen428 will be happy :D

File: Gemfile.lock [hunk 4]
```diff
@@ -842,7 +842,7 @@ GEM
 
 PLATFORMS
   arm64-darwin-20
```
Comment (joshpuetz) [inline Gemfile.lock:844]:
```suggestion
  arm64-darwin
```

Suggest removing the version number

File: Gemfile.lock [hunk 5]
```diff
@@ -842,7 +842,7 @@ GEM
...
-  x86_64-darwin-18 
+  x86_64-darwin
   x86_64-darwin-19
   x86_64-darwin-20
```
Comment (joshpuetz) [inline Gemfile.lock:847]:
```suggestion
```

Suggest removing these lines as we are covered by the line `x86_64-darwin`

**Review by joshpuetz — CHANGES_REQUESTED** [review]
A quick suggestion related to versions in the Gemfile.lock, otherwise looks good to me!

**Comment by rhymes** [comment]
@joshpuetz done, thank you!

**Review by maestromac — APPROVED** [review]

**Review by jgaskins — APPROVED** [review]
Nice! 🙌🏼

**Comment by rhymes** [comment]
There's a broken spec, because now that Rails enforces the check constraints, we need to adjust a data update script. I'm on it!

**Review by citizen428 — APPROVED** [review]

**Review by joshpuetz — APPROVED** [review]

**Review by joshpuetz — APPROVED** [review]

**Pull request merged**